src/synthetix/StakingRewards.t.sol

```diff
@@ -270,16 +270,40 @@ contract StakingRewardsTest is Test {
         assertEq(rewards.rewardsDuration(), 70 days);
     }
 
-    function testRevertSetRewardsDurationOnActiveDistribution() public {
+    function testSetRewardsDurationOnActiveDistribution() public {
         setupStakingToken(100 * WAD);
         rewards.stake(100 * WAD);
-
         setupReward(100 * WAD);
+        assertEq(rewardGem.balanceOf(address(this)), 0);
+        assertEq(rewardGem.balanceOf(address(rewards)), 100 * WAD);
+        assertEq(gem.balanceOf(address(this)), 0);
+        assertEq(gem.balanceOf(address(rewards)), 100 * WAD);
+        assertEq(rewards.rewardPerTokenStored(), 0);
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), 100 * WAD / 7 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 7 days);
+        assertEq(rewards.rewardsDuration(), 7 days);
 
         skip(1 days);
 
-        vm.expectRevert("Previous rewards period must be complete before changing the duration for the new period");
         rewards.setRewardsDuration(70 days);
+
+        assertEq(rewards.rewardPerTokenStored(), 1 days * (100 * WAD / 7 days) * WAD / (100 * WAD));
+        assertEq(rewards.lastUpdateTime(), block.timestamp);
+        assertEq(rewards.rewardRate(), 6 days * (100 * WAD / 7 days) / 70 days);
+        assertEq(rewards.periodFinish(), block.timestamp + 70 days);
+        assertEq(rewards.rewardsDuration(), 70 days);
+
+        skip(70 days);
+
+        rewards.exit();
+
+        uint256 rewardPaid = 1 days * (100 * WAD / 7 days) + 70 days * (6 days * (100 * WAD / 7 days) / 70 days);
+        uint256 rewardDust = 100 * WAD - rewardPaid;
```

Comment:
Might be good to also assert that `rewardDust` is really a negligible amount, so maybe verifying it's smaller than X days (maybe need to dive deeper but it seems the difference is due to dividing by days in a different order then in the contract logic).

Reply:
The dust is due to the rounding down in the `rewardRate` calculation. I rewrote the test a little bit to try and make this clearer.